File: daq/slc/src/dqm/src/IntArray.cc
#include "dqm/IntArray.h"

#include <sstream>

using namespace Belle2;

IntArray::IntArray() throw() : NumberArray(0), _value_v(NULL) {}

IntArray::IntArray(size_t size) throw() : NumberArray(), _value_v(NULL)
{
  resize(size);
}

IntArray::IntArray(const NumberArray& v) throw() : NumberArray(), _value_v(NULL)
{
  resize(v.size());
  for (size_t n = 0; n < size(); n++) set(n, _value_v[n]);
}

IntArray::~IntArray() throw()
{
  if (_value_v != NULL) delete [] _value_v;
}

double IntArray::get(size_t n) const throw()
{
  if (n < size()) return _value_v[n];
  else return -1;
}
void IntArray::set(size_t n, double v) throw()
{
  if (n < size()) _value_v[n] = (int)v;
}
void IntArray::resize(int s) throw()
{
  if (_value_v != NULL) delete [] _value_v;
  _value_v = new int [s];
  _size = s;
  for (size_t n = 0; n < size(); n++) set(n, 0);
}

std::string IntArray::toScript() const throw()
{
  std::stringstream ss("");
  for (size_t n = 0; n < size(); n++) {
    ss << get(n);
    if (n < size() - 1) ss << ",";
  }
  return ss.str();
}

int& IntArray::operator[](size_t n) throw()
{
  if (n < size()) return _value_v[n];
  return _value_v[0];
}

int IntArray::operator[](size_t n) const throw()
{
  if (n < size()) return _value_v[n];
  return 0;
}

void IntArray::readObject(Belle2::Reader& reader) throw(Belle2::IOException)
{
  for (size_t n = 0; n < size(); n++) {
    _value_v[n] = reader.readInt();
  }
}

void IntArray::writeObject(Belle2::Writer& writer) const throw(Belle2::IOException)
{
  for (size_t n = 0; n < size(); n++) {
    writer.writeInt(_value_v[n]);
  }
}

double IntArray::readElement(Belle2::Reader& reader) throw(Belle2::IOException)
{
  return reader.readInt();
}

void IntArray::writeElement(double value, Belle2::Writer& writer)
const throw(Belle2::IOException)
{
  writer.writeInt(value);
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/IntArray.java
package b2daq.dqm.core;

import java.io.IOException;

import b2daq.core.Reader;
import b2daq.core.Writer;

public class IntArray extends NumberArray {

	private int[] _value_v;
	
	public IntArray() {
		_value_v = new int [0];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public IntArray(int len) {
		_value_v = new int [len];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public IntArray(NumberArray array) { copy(array); }

	public void copy(NumberArray array) {
		if( _value_v.length != array.length() ) _value_v = new int [array.length()];
		for (int n = 0; n < length(); n++ ) _value_v[n] = (int)array.get(n);
	}

	public void copy(int [] array) {
		if( _value_v.length != array.length ) _value_v = new int [array.length];
		for (int n = 0; n < length(); n++ ) _value_v[n] = array[n];
	}

	public double get(int index) {
		if ( index >= 0 && index < length() ) return _value_v[index];
		else return -1;
	}

	public void set(int index, double data) {
		if ( index >= 0 && index < length() ) _value_v[index] = (int)data;
	}

	public void set(int index, int data) {
		if ( index >= 0 && index < length() ) _value_v[index] = data;
	}

	public int length() { return _value_v.length; }

	public void resize(int len) {
		_value_v = new int [len];
		for (int n = 0; n < length(); n++ ) _value_v[n] = 0;
	}

	public void writeSingle(double value, Writer writer) throws IOException {
		writer.writeInt((int)value);
	}

	public double readSingle(Reader reader) throws IOException {
		return reader.readInt();
	}

}
